### ISSUE TITLE
What is the function of  self.model.train() in /PyTorch/Translation/Transformer/fairseq/trainer.py ?

### ISSUE THREAD

**Issue by yaoyiran** [NVIDIA/DeepLearningExamples#148]
In trainer.py it seems that training loss is calculated by line 230:
loss, sample_size, logging_output_ = self.task.get_loss(self.model, self.criterion, sample).

However, when I go into line 225:  self.model.train(), I found that corresponds to line 87 in /Translation/Transformer/fairseq/models/fairseq_model.py: 

        def train(mode):
            if mode:
                raise RuntimeError('cannot train after make_generation_fast')
        # this model should no longer be used for training
        self.eval()
        self.train = train

Does "this model should no longer be used for training" mean that line 225:  self.model.train() in trainer.py is dumb and does not have any function?

**Comment by jbaczek** [comment]
This means that you should use fairseq trainer class for training transformer
